src/main/java/com/example/DataService/CurrencyService.java

```diff
@@ -0,0 +1,101 @@
+package com.example.DataService;
+
+
+import java.math.BigDecimal;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import com.example.repository.CurrencyRepository;
+import com.example.repository.HistoryRepository;
+
+import lombok.RequiredArgsConstructor;
+import model.Asset;
+import model.Currency;
+import model.History;
+
+@RequiredArgsConstructor
+@EnableScheduling
+@Service
+public class CurrencyService {
+	
+	@Value("${my.url}")
+	private String uri;
+	
+	final CurrencyRepository cr;
+	final HistoryRepository hr;
+
+	private int constraint;
```

Comment:
why isn't this a local variable?

Reply:
Because it was used in two methods, putCurrencies() and updateHistory().